### PR TITLE
Move to Hatch

### PR DESCRIPTION
**Pull request by xoudini**
Basically the title. See [Hatch](https://github.com/pypa/hatch) docs for more details.

### CODE REVIEW

**Comment by occamz** [comment]
Oh cool, thank you!

I haven't tried Hatch yet, but I guess this is a good opportunity. I will look at this and the Ruff PR at some point later this week :)

**Comment by xoudini** [comment]
> Oh cool, thank you!

No problemo!

> I haven't tried Hatch yet, but I guess this is a good opportunity.

A good quickstart is:

- `hatch fmt` runs ruff linter and formatter in one (use `--formatter` or `--linter` to run them separately)
- `hatch run types:check` for type checking (PR incoming)
- `hatch test` for tests (although there aren't any at the moment)

And the [versioning](https://hatch.pypa.io/1.9/version) docs might be worth reading separately 🙂  

> I will look at this and the Ruff PR at some point later this week :)

It's pretty tiny, even with the formatting/linting updates, but looks larger since it technically should target this branch, but thanks!

File: pyproject.toml [hunk 1]
```diff
@@ -1,28 +1,81 @@
...
 name = "docker-snapshot"
-homepage = "https://github.com/occamz/ds"
-documentation = "https://github.com/occamz/ds"
-version = "1.0.11"
+dynamic = ["version"]
 description = "`ds` is a development utility for managing snapshots inside a docker container."
-authors = ["occamz"]
 readme = "README.md"
+requires-python = ">=3.9"
+license = "Apache-2.0"
```
Comment (xoudini) [inline pyproject.toml:12]: Oh right. I absent-mindedly committed this with my defaults, but just noticed that there isn't a license to begin with. Thoughts on this, or should I replace this with `"Other/Proprietary License"` for now?

File: pyproject.toml [hunk 2]
```diff
@@ -1,28 +1,81 @@
...
+keywords = []
+authors = [{ name = "occamz" }]
+classifiers = [
+    "Development Status :: 4 - Beta",
```
Comment (xoudini) [inline pyproject.toml:16]: Also might want to comment on this (e.g. see [the full list](https://pypi.org/classifiers) if you want this to be something else).

File: pyproject.toml (hunk 1 above)
Comment (occamz) [reply pyproject.toml:12]: Apache 2 is fine, I am not too well versed at OSS licences at this point, but seems sane from reading up on it for 5 minutes :D

File: pyproject.toml (hunk 2 above)
Comment (occamz) [reply pyproject.toml:16]: Same as above, I think it is fine.

File: pyproject.toml (hunk 1 above)
Comment (xoudini) [reply pyproject.toml:12]: Cool, should I add the license file while I'm at it?

Comment (occamz) [reply pyproject.toml:12]: Go for it 👍

Comment (xoudini) [reply pyproject.toml:12]: Done ✅

**Review by occamz — APPROVED** [review]
Looks good!

**Pull request merged**